src/Hocon.Configuration/Config.cs

```diff
@@ -173,43 +143,46 @@ public virtual Config GetConfig(HoconPath path)
         public virtual Config WithFallback(Config fallback)
         {
             if (ReferenceEquals(fallback, this))
-                throw new ArgumentException("Config can not have itself as fallback", nameof(fallback));
+                return this; // no-op
 
             if (fallback.IsNullOrEmpty())
                 return this; // no-op
 
             if (IsEmpty)
                 return fallback;
 
-            return new Config(this, fallback);
+            var result = new Config(this);
+            result.MergeConfig(fallback);
```

Comment:
So this just appends the latest fallback to what position on the linked list?

Reply:
fallbacks are always added to the back of the chain